app/fruits/[fruitId]/actions.js

```diff
@@ -0,0 +1,26 @@
+'use server';
+
+import { cookies } from 'next/headers';
+import { getCookie } from '../../../util/cookies';
+import { parseJson } from '../../../util/json';
+
+export async function createComment(fruitId, comment) {
```

Comment:
maybe `createOrUpdateComment` to be more clear that this function has more that it does?

```suggestion
export async function createOrUpdateComment(fruitId, comment) {
```